### PR TITLE
Return error if underlying is not found.

### PR DESCRIPTION
**Pull request by gislenius**
fixes 28msec/cellstore-nttdata#38

### CODE REVIEW

**Comment by gislenius** [comment]
This is also pending the RESTXQ refactoring, as these errors won't be returned if some clauses get materialized because sequential.

**Comment by wcandillon** [comment]
@gislenius Who should review this?

**Comment by gislenius** [comment]
@dknochen this is now ready for your attention!

File: queries/public/api/components.jq [hunk 1]
```diff
@@ -244,4 +250,8 @@ let $serializers := {
 }
 
 let $results := api:serialize($result, $comment, $serializers, $format, "components")
-return api:check-and-return-results($token, $results, $format)
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else api:check-and-return-results($token, $results, $format)
```
Comment (dknochen) [inline queries/public/api/components.jq]: can we make it a switch?

File: queries/public/api/facttable-for-component.jq [hunk 2]
```diff
@@ -222,4 +231,10 @@ let $serializers := {
...
     }
 }
-return api:serialize($result, $comment, $serializers, $format, "facttable-" || $cid)
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else if($components-not-found)
+                 then api:not-found("component")
+                 else api:serialize($result, $comment, $serializers, $format, "facttable-" || $cid)
```
Comment (dknochen) [inline queries/public/api/facttable-for-component.jq]: can we make it a switch?

File: queries/public/api/labels.jq [hunk 3]
```diff
@@ -130,4 +136,8 @@ let $serializers := {
 }
 
 let $results := api:serialize($result, $comment, $serializers, $format, "labels")
-return api:check-and-return-results($token, $results, $format)
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else api:check-and-return-results($token, $results, $format)
```
Comment (dknochen) [inline queries/public/api/labels.jq]: can we make it a switch?

File: queries/public/api/modelstructure-for-component.jq [hunk 4]
```diff
@@ -196,9 +204,10 @@ let $serializers := {
...
-        { status: 404 },
-        session:error("component not found", "json")
-    }
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else if($components-not-found)
+                 then api:not-found("component")
+                 else api:serialize($result, $comment, $serializers, $format, "components")
```
Comment (dknochen) [inline queries/public/api/modelstructure-for-component.jq]: can we make it a switch?

File: queries/public/api/periods.jq [hunk 5]
```diff
@@ -90,4 +95,8 @@ let $serializers := {
 }
 
 let $results := api:serialize($result, $comment, $serializers, $format, "filings")
-return api:check-and-return-results($token, $results, $format)
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else api:check-and-return-results($token, $results, $format)
```
Comment (dknochen) [inline queries/public/api/periods.jq]: can we make it a switch?

File: queries/public/api/report-elements.jq [hunk 6]
```diff
@@ -224,4 +230,8 @@ let $serializers := {
     }
 }
 let $results := api:serialize($result, $comment, $serializers, $format, "report-elements")
-return api:check-and-return-results($token, $results, $format)
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else api:check-and-return-results($token, $results, $format)
```
Comment (dknochen) [inline queries/public/api/report-elements.jq]: can we make it a switch?

File: queries/public/api/spreadsheet-for-component.jq [hunk 7]
```diff
@@ -128,4 +137,10 @@ let $results :=
...
             $spreadsheet
         }
-return api:check-and-return-results($token, $results, $format)
+return if($entities-not-found)
+       then api:not-found("entity")
+       else if($archives-not-found)
+            then api:not-found("archive")
+            else if($components-not-found)
+                 then api:not-found("component")
+                 else api:check-and-return-results($token, $results, $format)
```
Comment (dknochen) [inline queries/public/api/spreadsheet-for-component.jq]: can we make it a switch?

**Comment by dknochen** [comment]
Very nice :) I marked several places where we could improve the code quality.

Additionally, we should add the error code 404 to relevant endpoints in swagger/queries.json.

I believe, its good to go then. thank you!

**Comment by gislenius** [comment]
@dknochen There you go!

**Comment by dknochen** [comment]
:+1: beautiful

**Pull request merged**